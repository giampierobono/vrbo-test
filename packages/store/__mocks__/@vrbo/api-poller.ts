// tslint:disable-next-line:no-var-requires
const apiPollerStore = require("../../../../__mocks__/@vrbo/api-poller");

module.exports = apiPollerStore;
