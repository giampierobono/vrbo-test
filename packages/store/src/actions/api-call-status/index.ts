export * from "./api-call-status.actions";
