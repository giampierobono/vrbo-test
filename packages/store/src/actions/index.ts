export * from "./api-call-status";
export * from "./poller-config";
export * from "./properties-sort";
export * from "./property";
