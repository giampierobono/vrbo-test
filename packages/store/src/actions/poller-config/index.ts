export * from "./poller-config.actions";
