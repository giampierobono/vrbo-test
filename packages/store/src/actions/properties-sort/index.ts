export * from "./properties-sort.actions";
