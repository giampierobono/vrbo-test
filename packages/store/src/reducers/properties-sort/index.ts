export * from "./properties-sort.reducer";
