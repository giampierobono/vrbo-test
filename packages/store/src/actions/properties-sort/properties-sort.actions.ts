import { ActionWithPayload, PropertiesSortByConfig } from "@vrbo/data-models";

export enum PropertiesSortActions {
  SetSortByConfig = "[PropertiesSort] Set new sort param"
}

export const setNewPropertiesSortAction = (
  sortBy: PropertiesSortByConfig
): ActionWithPayload<PropertiesSortByConfig> => ({
  type: PropertiesSortActions.SetSortByConfig,
  payload: sortBy
});
