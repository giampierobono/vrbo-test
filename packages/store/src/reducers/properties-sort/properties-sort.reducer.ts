import {
  ActionWithPayload,
  PropertiesSortBy,
  PropertiesSortByConfig
} from "@vrbo/data-models";
import { PropertiesSortActions } from "../../actions/properties-sort";

export interface PropertiesListSortByState {
  sortConfig: PropertiesSortByConfig;
}

export const initialPropertiesListSortByState: PropertiesListSortByState = {
  sortConfig: {
    prop: PropertiesSortBy.Title,
    asc: true
  }
};

export const propertiesListSortReducer = (
  state: PropertiesListSortByState = initialPropertiesListSortByState,
  action: ActionWithPayload<PropertiesSortByConfig>
): PropertiesListSortByState => {
  switch (action.type) {
    case PropertiesSortActions.SetSortByConfig: {
      return {
        ...state,
        sortConfig: { ...action.payload! }
      };
    }
    default: {
      return state;
    }
  }
};
