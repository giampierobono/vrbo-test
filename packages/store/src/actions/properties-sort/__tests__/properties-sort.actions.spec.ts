import { PropertiesSortBy } from "@vrbo/data-models";
import { setNewPropertiesSortAction } from "../properties-sort.actions";

describe("Properties sort config actions", () => {
  it("should return correct setNewPropertiesSortAction object", () => {
    expect(
      setNewPropertiesSortAction({ prop: PropertiesSortBy.Title, asc: true })
    ).toMatchSnapshot();
  });
});
