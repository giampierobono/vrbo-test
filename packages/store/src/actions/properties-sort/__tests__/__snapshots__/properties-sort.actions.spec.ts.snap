// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Properties sort config actions should return correct setNewPropertiesSortAction object 1`] = `
Object {
  "payload": Object {
    "asc": true,
    "prop": "title",
  },
  "type": "[PropertiesSort] Set new sort param",
}
`;
